src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,61 @@
 package core.basesyntax;
 
+import java.text.ParseException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    public static final String REPORT_TITLE_PATTERN = "Report for period %s - %s";
+    public static final String PERSON_DESCRIPTION_PATTERN = "%s - %s";
+    public static final String DATA_SEPARATOR = " ";
+    public static final int DATE_POSITION = 0;
+    public static final int NAME_POSITION = 1;
+    public static final int HOURS_POSITION = 2;
+    public static final int INCOME_POSITION = 3;
+    private static final DateTimeFormatter D_FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        final LocalDate from = LocalDate.parse(dateFrom, D_FORMATTER);
+        final LocalDate to = LocalDate.parse(dateTo, D_FORMATTER);
+        final String title = String.format(REPORT_TITLE_PATTERN, dateFrom, dateTo);
+        final StringBuilder builder = new StringBuilder(title)
+                .append(System.lineSeparator());
+        int salary = 0;
+        int separatorIndex = names.length - 1;
+        for (String name : names) {
+            for (String dataElement : data) {
+                String[] splitLine = dataElement.split(DATA_SEPARATOR);
+                try {
+                    if (splitLine[NAME_POSITION].equals(name)
+                            && isBetween(from, to, splitLine[DATE_POSITION])) {
+                        salary += getSalary(splitLine[INCOME_POSITION], splitLine[HOURS_POSITION]);
+                    }
+                } catch (ParseException e) {
+                    e.printStackTrace();
```

Comment:
e.printStackTrace() is a bad practice))
better throw an exception

Reply:
Yes, I found out about it in the Files topic. Bogdan does not stop repeating. I'll fix it :)